### DISCUSSION TITLE
How to enhance / augment features in MarkdownEditor

### DISCUSSION THREAD

**Discussion by ZehuaZhang** [primer/react#3225] [Q&A] [answered]
The `MarkdownEditor` component is great like having drag and drop and upload files, really impressive.

I was wondering how I could add some extensions and enhancements to the markdown text itself, like
* `remark-gfm`: github flavored markdown
* `mermaid`: diagram
* `katex`: math equation

**Comment by colebemis** [comment, marked as answer]
Hi, @ZehuaZhang! `MarkdownEditor` intentionally leaves markdown parsing up to the consumer of the component. So you can use `remark` (with any plugins you need) directly, then pass the output HTML to `MarkdownEditor`. @iansan5653 is that right?

**Reply by iansan5653** [reply]
Yep, that's right! It accepts as a prop any async function that can turn Markdown into HTML. You can also render custom toolbar buttons if you want to add, for example, a shortcut to add a math equation.

**Reply by ZehuaZhang** [reply]
Do you guys have some examples to share?
